sklearn/tests/test_config.py

```diff
@@ -0,0 +1,26 @@
+from sklearn import get_config, set_config
+from sklearn.utils.testing import assert_equal
+
+
+def test_set_config():
+    assert_equal(get_config(), {'assume_finite': False})
+
+    # Not using as a context manager affects nothing
+    set_config(assume_finite=True)
```

Comment:
this seems surprising behavior.
How about we use a naming scheme similar to matplotlib?
http://matplotlib.org/users/style_sheets.html

maybe ``config_context``? In particular you currently provide no way to set this globally apart from writing to the private constant, which people will definitely do ;)

tldr; I expect something that's called ``set_config`` to set the config